src/devices/LidarLiteV3/LidarLiteV3.cs

```diff
@@ -0,0 +1,319 @@
+    /// </summary>
+    public class LidarLiteV3 : IDisposable
+    {
+        /// <summary>
+        /// Default address for LidarLiteV3
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x62;
+
+        private GpioController _gpioController;
+        private I2cDevice _i2cDevice;
+        private int? _powerEnablePin;
+
+        /// <summary>
+        /// Initialize the LidarLiteV3
+        /// </summary>
+        /// <param name="i2cDevice">I2C device</param>
+        /// <param name="gpioController">GPIO controller</param>
+        /// <param name="powerEnablePin">The pin number used to control power to the device</param>
+        public LidarLiteV3(I2cDevice i2cDevice, GpioController gpioController = null, int? powerEnablePin = null)
+        {
+            _gpioController = gpioController;
+            _i2cDevice = i2cDevice;
+            _powerEnablePin = powerEnablePin;
+
+            if (_gpioController != null && _powerEnablePin.HasValue)
+            {
+                _gpioController.OpenPin(_powerEnablePin.Value, PinMode.Output);
+                PowerOn();
+            }
+
+            Reset();
+        }
+
+        /// <summary>
+        /// Power off the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOff()
+        {
+            if (_gpioController != null && _powerEnablePin.HasValue)
+            {
+                _gpioController.Write(_powerEnablePin.Value, PinValue.Low);
+            } else
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+        }
+
+        /// <summary>
+        /// Power on the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOn()
+        {
+            if (_gpioController != null && _powerEnablePin.HasValue)
+            {
+                _gpioController.Write(_powerEnablePin.Value, PinValue.High);
+            }
+            else
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+        }
+
+        /// <summary>
+        /// Reset FPGA, all registers return to default values
+        /// </summary>
+        public void Reset()
+        {
+            WriteRegister(Register.ACQ_COMMAND, 0x00);
+        }
+
+        /// <summary>
+        /// Measure distance in cm
+        /// </summary>
+        /// <remarks>
+        /// Note: Do not call if running while in repetition mode.  It will block until 
+        /// repetition finishes (forever if infinite).
+        /// </remarks>
+        /// <param name="withReceiverBiasCorrection">Faster without bias correction, but more prone to errors if condition changes.</param>
+        /// <returns>Distance in cm</returns>
+        public int MeasureDistance(bool withReceiverBiasCorrection = true)
```

Comment:
This could actually be an Async method (note: you should also provide synchronous version which can call async).

You can use `await Task.Delay(1)` instead of sleep and rest would remain the same (except signature)

Reply:
will fix